hail/src/main/scala/is/hail/types/physical/stypes/concrete/SNDArrayPointer.scala

```diff
@@ -45,16 +45,23 @@ object SNDArrayPointerSettable {
 
 class SNDArrayPointerSettable(val st: SNDArrayPointer, val a: Settable[Long]) extends PNDArrayValue with PSettable {
   val pt: PNDArray = st.pType
+  private var shape: IndexedSeq[Value[Long]] = null
+  private var strides: IndexedSeq[Value[Long]] = null
+  private var dataFirstElement: Settable[Long] = null
```

Comment:
Is this an ok thing to do Tim? You avoid the nulls when saving the length of arrays by passing in an uninitialized variable. I could do that too, but honestly I'd rather get an NPE at hail compile time than be confused about provenance of a 0 pointer at run time.

Reply:
the design you've got here (create the variables in `store`) slightly breaks the mental model of settables, because they should be thought of as a single variable or set of variables which you can assign to multiple times.

I think it's possible to write down an example of code generation where this actually breaks the interface of settables, because you end up with loads referencing the wrong (old) variables.

Reply:
yes, here:

```

val nd = new NDArrayPointerSettable(...)

cb.assign(nd, someInitialValue)
cb.whileLoop(someBoolean(nd.load(), {
  
  // do some work
  cb.assign(nd, someNewValue)
})
```

The `nd.load()` in the condition of the while loop references different variables than the ones stored to on the last line. We have to use the same vars.

Reply:
I think I see what you're trying to do, though -- you can just set a `private var storedAtLeastOnce: Boolean = false`, set that to true in `store`, and assert it's true in `load`.

Reply:
Resolved this by changing to match the pattern of `SIndexablePointerSettable`. Decided it was less complicated to do the same thing in both places.